### PR TITLE
Added SSL related fields to access.log

### PR DESCRIPTION
**Pull request by acaciocenteno**
Added three fields to access log:
- SSL version, for instance 'TLSv1/SSLv3'.
- SSL cipher suite, for instance 'ECDHE-RSA-AES128-GCM-SHA256'
- SSL session reused.

### CODE REVIEW

**Comment by jpeach** [comment]
FWIW for this specific case, you can use the ``sslheaders`` plugin as a workaround. Put the values you want into HTTP headers, then log those. In fact, maybe using ``@headers`` is a reasonable solution to @SolidWallOfCode's logging data problem.

**Comment by bryancall** [comment]
There is a related Jira for this:
Desire support for client TLS cipher in custom log format
https://issues.apache.org/jira/browse/TS-3648

File: iocore/net/P_SSLNetVConnection.h [hunk 1]
```diff
@@ -120,6 +120,28 @@ class SSLNetVConnection:public UnixNetVConnection
     sslClientRenegotiationAbort = state;
   };
 
+  const char * get_ssl_protocol(void) const
+  {
+    if ( ssl == NULL )
+      return NULL;
+    return SSL_get_cipher_version(ssl);
```
Comment (shinrich) [inline iocore/net/P_SSLNetVConnection.h]: Agreed, SSL_get_version is probably what you want here.

**Comment by shinrich** [comment]
@fpesce got the ssl_session_reuse in TS-3683 (cqssl rather than cssr for the log field).  It also includes session reuse via tickets.

Being able to record the the selected cipher and negotiated protocol would still be valuable.  I see that @ericcarlschwartz is working on the related bug TS-3648.  Hopefully he can pull these changes in via that issue.

**Pull request closed without merging**